plugins/kubernetes-backend/src/service/KubernetesBuilder.ts

```diff
@@ -325,4 +340,28 @@ export class KubernetesBuilder {
 
     return objectTypesToFetch;
   }
+
+  protected async makeProxyRequest(
+    req: express.Request,
+    res: express.Response,
+  ) {
+    const services = this.getProxyServices();
+    const proxyResponse: KubernetesProxyResponse =
+      await this.proxy.handleProxyRequest(services, req);
+    res.status(proxyResponse.code).json(proxyResponse.data);
+  }
+
+  protected getProxyServices(): KubernetesProxyServices {
+    const kcs =
+      this.clusterSupplier ??
+      this.buildClusterSupplier(this.defaultClusterRefreshInterval);
+
+    if (!kcs) {
+      this.env.logger.error('could not find cluster supplier!');
```

Comment:
did you really mean to just log and continue here, or should it throw?

Reply:
Huh, I think I left this log in because I wasn't sure entirely what to do if we couldn't even find a cluster supplier. What kind of error would make the most sense to throw here?

Reply:
I'd say either make it impossible to happen (maybe that's already the case here?) or make it a 500 (just throwing an Error)

Reply:
I can't see this happening since as far as I can tell `buildClusterSupplier` will either return an array (empty or with data) or throw an error, `null` check is nice to have just to be safe.

Reply:
To be specific, it should throw when `clusterLocatorMethods` is missing from config, when the `type` of any cluster is not supported, or when the `getClusters()` method throws, which depends on which provider type is configured.